### ISSUE TITLE
[API] Filtre de dates pour les metrics n'ont plus d'effet sur metrics retournées

### ISSUE THREAD

**Issue by alexphiev** [dataforgoodfr/12_bloom#286]
Il y a eu des changements sur les endpoint metrics et maintenant les valeurs de durées retournées ne respectent pas le filtre de date appliqué.

Les trois endpoints sont affectés :
- /metrics/zones/660/visiting-time-by-vessel sur => `zone_visiting_time_by_vessel` ne change pas en fonction des dates
- /metrics/zones-visited => `visiting_duration` ne change pas en fonction des dates
- /metrics/vessels-in-activity => `total_time_at_sea` ne change pas en fonction des dates

**Comment by louwelgryn** [comment]
Est-ce que vous pouvez préciser les changements qui ont été faits ? et pourquoi toucher aux durées ? Merci beaucoup !

**Comment by rv2931** [comment]
D'une manière générale quand tu vois ce genre de chose, essaie de vérifier si ce n'est pas du au cache. Je ne sais pas si tu l'as fait mais si tu mets "nocache=true" (je vais changer le nom parce que là ça fait très Windows inversation de la logique inversée... ^^)
Si lorsque nocache=true le problème persiste alors ça confirme que le souci viens de la requête, si ce n'est pas le cas alors ça veut dire que ça vient de la gestion du cache.
J'ai pas regardé pour ces endpoints je suis en train de recharger un dump à jour pour l'issue de l'erreur 500, je pensais pas que ça prendrait autant de temps :/

**Comment by louwelgryn** [comment]
j'ajoute ici que quand on change la date - ça prend plusieurs dizaines de secondes à charger ... c'est un peu lent je sais pas si on peut optimiser ça ?

**Comment by rv2931** [comment]
J'ai regardé un peu la BDD issue d'un dump de la prod de la semaine dernière
Soit les données sont bizarres, soit j'ai pas compris quelles données utiliser
Dans fct_segment tous les segments ont une date timestamp_start supérieure à 2024-11-19 15:05:27+01
Donc en l'état aucune zone ne peut donc avoir été traversée avant le 19/11/2024
Est-ce que vous confirmez les mêmes données dans fct_segment ?
Je vais recharger un dump de la production pour voir si c'est pas mes données qiu sont corrompues

**Comment by njouanin** [comment]
oui, bizarre on dirait que toutes les relations segment/zones avant le 19/11 ont été supprimées...
Pour moi [la dernière correction date d'il y a 1 mois](https://github.com/dataforgoodfr/12_bloom/issues/234) lorsque j'ai delete quelques relations, mais pas à ce point là !

**Comment by njouanin** [comment]
En exécutant la requête sur une base datant du 22/10 je retrouve plus de 412000 segments associés à des zones, les plus anciens datant du 02/04/2024. 
En prod on en a actuellement que 23000 tous > au 19/11.
=> _Houston, we have a problem_

**Comment by rv2931** [comment]
:/
Bon ça dit potentiellement que c'est pas forcément un problème de filtre de dates mais un problème de données... Maintenant c'est arrivé et comment on le récupère/régénère...???

**Comment by rv2931** [comment]
Je passe cet issue en standby le temps de régler le problème de donnée que j'ai remonté sur un autre ticket #299

**Comment by rv2931** [comment]
Je viens de restaurer les relations segment/zone jusqu'au 30/10 à partir d'un dump de marthe
Ce serait possible de revalider le souci de cet issue maintenant qu'on a des données plus complète. Après il y a d'autres soucis déjà pré existants sur les associations segment/zone qui seront corrigé plus tard
En attente de vos retours

**Comment by marthevienne** [comment]
C'est bon pour moi. Il manque quand même des données sur les relations entre le 30/10 et le 19/11 mais on a le comportement qu'on attend.
